src/NavBar.php

```diff
@@ -47,28 +47,38 @@ public function begin(): ?string
         $navOptions = $this->options;
         $navTag = ArrayHelper::remove($navOptions, 'tag', 'nav');
 
-        if (!is_string($navTag) && !is_bool($navTag) && $navTag !== null) {
-            throw new InvalidArgumentException('Tag should be either string, bool or null.');
+        if (
+            !is_string($navTag) &&
+            $navTag !== '' &&
+            !is_bool($navTag) &&
+            $navTag !== null
+        ) {
+            throw new InvalidArgumentException('Tag should be either non empty string, bool or null.');
         }
 
         return
-            Html::beginTag($navTag, $navOptions) . "\n" .
+            (is_string($navTag) ? Html::openTag($navTag, $navOptions) : '') . "\n" .
             $this->brand . "\n" .
-            Html::beginTag('div', $this->menuOptions) .
-            Html::beginTag('div', $this->itemsOptions);
+            Html::openTag('div', $this->menuOptions) .
+            Html::openTag('div', $this->itemsOptions);
     }
 
     protected function run(): string
     {
         $tag = ArrayHelper::remove($this->options, 'tag', 'nav');
-        if (!is_string($tag) && !is_bool($tag) && $tag !== null) {
-            throw new InvalidArgumentException('Tag should be either string, bool or null.');
+        if (
+            !is_string($tag) &&
+            $tag !== '' &&
```

Comment:
Second condition doesn't make sense. If value isn't string it's not empty string by default.

Reply:
Fixed and add tests.